src/ZoneServer/World/Actors/CombatEntities/Components/MovementComponent.cs

```diff
@@ -125,28 +127,79 @@ private TimeSpan MoveToConditional(Position destination, bool executeMove)
 				if (distance <= 10)
 					return TimeSpan.Zero;
 
-				// Calculate movement and move time
-				_moveTime = TimeSpan.FromSeconds(distance / speed);
-				_moveX = (diffX / _moveTime.TotalSeconds);
-				_moveZ = (diffZ / _moveTime.TotalSeconds);
+				// Use Pathfinder to find a path to the destination
+				var pathfinder = this.Entity.Map.Pathfinder;
+				_path = pathfinder.FindPath(position, destination, this.Entity.EffectiveSize);
+
+				// Don't move if no path found
+				if (_path.Count == 0)
+					return TimeSpan.Zero;
 
+				// Calculate total move time to destination
+				double totalDistance = 0;
+				for (var i = 0; i < _path.Count - 1; i++)
+				{
+					totalDistance += _path[i].Get2DDistance(_path[i + 1]);
+				}
+				var totalMoveTime = TimeSpan.FromSeconds(totalDistance / speed);
+
+				// Execute Movement
 				if (executeMove)
 				{
 					this.IsMoving = true;
-					this.Destination = destination;
+					this.Destination = _path.Last();
 					this.MoveTarget = MoveTargetType.Position;
 
-					// Set direction relative to current position
-					this.Entity.Direction = position.GetDirection(destination);
+					this.MoveToNextPosition();
+				}
 
-					var fromCellPos = this.Entity.Map.Ground.GetCellPosition(this.Entity.Position);
-					var toCellPos = this.Entity.Map.Ground.GetCellPosition(this.Destination);
+				return totalMoveTime;
+			}
+		}
 
-					// Update clients
-					Send.ZC_MOVE_PATH(this.Entity, fromCellPos, toCellPos, speed);
-				}
+		/// <summary>
+		/// Sets the target of entity to the next position
+		/// in its path list.
+		/// </summary>
+		private void MoveToNextPosition()
+		{
+			if (_path.Count == 0)
+				return;
+
+			// Remove completed path node
+			_path.RemoveAt(0);
```

Comment:
Is there a particular reason you aren't using a Queue for the path?

Reply:
Yes - I'm stupid :)

Initially I was using a list and iterating over it, figured out that wasn't a good idea and didn't realize I basically had a Queue.